scripts/SANS/sans/algorithm_detail/mask_workspace.py

```diff
@@ -118,23 +118,37 @@ def mask_with_mask_files(mask_info, workspace):
         load_options = {"Instrument": idf_path,
                         "OutputWorkspace": EMPTY_NAME}
         load_alg = create_unmanaged_algorithm(load_name, **load_options)
+        dummy_params = {"OutputWorkspace": EMPTY_NAME}
+        mask_alg = create_unmanaged_algorithm("MaskInstrument", **dummy_params)
+        clear_alg = create_unmanaged_algorithm("ClearMaskedSpectra", **dummy_params)
 
         # Masker
-        mask_name = "MaskDetectors"
-        mask_options = {"ForceInstrumentMasking": True}
-        mask_alg = create_unmanaged_algorithm(mask_name, **mask_options)
         for mask_file in mask_files:
             mask_file = find_full_file_path(mask_file)
 
             # Get the detector ids which need to be masked
             load_alg.setProperty("InputFile", mask_file)
             load_alg.execute()
             masking_workspace = load_alg.getProperty("OutputWorkspace").value
-            # Mask the detector ids on the original workspace
-            mask_alg.setProperty("Workspace", workspace)
-            mask_alg.setProperty("MaskedWorkspace", masking_workspace)
+            # Could use MaskDetectors directly with masking_workspace but it does not
+            # support MPI. Use a three step approach via a, b, and c instead.
+            # a) Extract detectors to mask from MaskWorkspace
+            det_ids = []
+            for i in range(masking_workspace.getNumberHistograms()):
```

Comment:
Could this become costly for ESS instruments? Would it make sense to have this in a C++ function and export it to Python in the future?

Reply:
You are right, this actually has non negligible cost --- even for SANS2D it takes almost 0.2 seconds! I will see if I find a better solution.

Reply:
Pushed update is around 20x faster!